File: react-native-zebra-bridge/android/src/main/java/com/ezcorp/ezroam/zebraBridge/ZebraBridgePackage.java
package com.ezcorp.ezroam.zebraBridge;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import com.ezcorp.ezroam.zebraBridge.di.DependencyFactory;
import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.facebook.react.bridge.JavaScriptModule;

public class ZebraBridgePackage implements ReactPackage {
    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        ZebraBridgeModule module = DependencyFactory.INSTANCE.getZebraModule(reactContext);
        return Arrays.<NativeModule>asList(module);
    }

    public List<Class<? extends JavaScriptModule>> createJSModules() {
        return null;
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}

File: react-native-zebra-bridge/android/src/main/java/com/ezcorp/ezroam/zebraBridge/ZebraBridgePackage.kt
package com.ezcorp.ezroam.zebraBridge

import android.util.Log
import com.ezcorp.ezroam.zebraBridge.di.ApplicationSingleton
import com.ezcorp.ezroam.zebraBridge.di.ApplicationSingleton.initializeKoin
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.JavaScriptModule
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager
import org.koin.core.Koin
import org.koin.core.KoinComponent
import java.util.*
import org.koin.core.inject

class ZebraBridgePackage : ReactPackage, KoinComponent {

    private val bridgeModule by inject<ZebraBridgeModule>()

    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        Log.i("Package", "Creating native modules")
        initializeKoin(reactContext)
        return Arrays.asList<NativeModule>(bridgeModule)
    }

    fun createJSModules(): List<Class<out JavaScriptModule?>> {
        return listOf()
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return emptyList()
    }


    override fun getKoin(): Koin {
        return ApplicationSingleton.koinApplication!!.koin
    }
}